src/caching/github-actions-cache.ts

```diff
@@ -345,33 +351,48 @@ export class GitHubActionsCache implements Cache {
    * Otherwise return false.
    */
   private _maybeHandleServiceDown(
-    res: http.IncomingMessage,
+    res: Result<http.IncomingMessage, Error>,
     script: ScriptReference
-  ): boolean {
+  ): res is {ok: false; error: Error} {
     if (this._serviceIsDown) {
       return true;
```

Comment:
in this case do we know that `res` is `{ok: false, error: Error}`?

Reply:
No, good point. It was also slightly wrong for the other cases, since we also handled the case where ok was true, but there were certain HTTP error codes.

Removed this case, since it was only there to prevent logging, and inverted the return value so that we can correctly say that if we return `true`, then `ok` must be true (which is true!).